src/game_player.cpp

```diff
@@ -180,6 +180,30 @@ void Game_Player::UpdateScroll(int old_x, int old_y) {
 	int new_panx = new_x - screen_x;
 	int new_pany = new_y - screen_y;
 
+	// Detect whether we crossed map boundary.
+	// We need to scale down dx/dy to a single step
+	// to not message up further calculations.
+	// FIXME: This logic will break if something moves so fast
+	// as to cross half the map in 1 frame.
```

Comment:
So this will break when you are in a Vehicle and "SetVehiclePosition" is used to jump across half the map?

Reply:
Well the answer is.. Yes and No.

It doesn't break in the way you're thinking. SetVehiclePosition() uses MoveTo() which doesn't use this scroll logic. 

The problem is, `MoveTo()` needs to reset the panorama. This is what happens when you use the vehicle trick or transport yourself to the same map.

Unfortunately, we call `MoveTo()` when loading saved games, so I can't make that change now because it will break save games loading.

We need commit e468f3f8d1ab4a3d884d38cb85 from #1522. This commit removes the call to `MoveTo()` from the load game path. Once that's done I can augment `MoveTo()` to reset the panorama.

Not resetting the panorama when you move your vehicle is a minor glitch, so I think I can add that part after this and #1522 are merged.